### PR TITLE
[JBPM-9548] Need access to process variables tags at FormRendererBase…

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9548

### CODE REVIEW

**Comment by elguardian** [comment]
Jenkins retest this please

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/BPMN2DataServiceImpl.java [hunk 1]
```diff
@@ -159,6 +159,9 @@ private ProcessAssetDesc fillProcessDefinition(ProcessDescriptor helper, KieCont
 
         definition.setAssociatedEntities(helper.getTaskAssignments());
         definition.setProcessVariables(helper.getInputs());
+        for(String varName : helper.getInputs().keySet()) {
+            definition.addTagsForVariable(varName, definition.getTagsForVariable(varName));
```
Comment (afalhambra) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/BPMN2DataServiceImpl.java:163]: I believe we should call `helper.getInputsTags()` instead here
```suggestion
            definition.addTagsForVariable(varName, helper.getInputTags(varName));
```

**Review by afalhambra — CHANGES_REQUESTED** [review]
Some changes required. 
Also a test case will be useful to include here for the method `getTagsForVariable` in the `ProcessDefinition` class

File: jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/model/ProcessDefinition.java [hunk 2]
```diff
@@ -137,4 +137,10 @@
      * @return @{@link TimerDesc} timers
      */
     Set<TimerDesc> getTimers();
+
+    /**
+     * Return the list of tags for certain process variable
+     * @return
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/model/ProcessDefinition.java:143]: done

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/ProcessAssetDesc.java [hunk 3]
```diff
@@ -321,6 +325,17 @@ public void setTimers(Set<TimerDesc> timers) {
         return timers;
     }
 
+    public void addTagsForVariable(String varName, Set<String> tags) {
+        if(!tagsByVariable.containsKey(varName)) {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/ProcessAssetDesc.java:329]: done

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/ProcessAssetDesc.java [hunk 4]
```diff
@@ -321,6 +325,17 @@ public void setTimers(Set<TimerDesc> timers) {
...
+            tagsByVariable.put(varName, new HashSet<>());
+        }
+        tagsByVariable.get(varName).addAll(tags);
+    }
+    @Override
+    public Set<String> getTagsForVariable(String varName) {
+        return tagsByVariable.containsKey(varName) ? tagsByVariable.get(varName) : emptySet();
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/ProcessAssetDesc.java:336]: done

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/BPMN2DataServiceImpl.java (hunk 1 above)
Comment (elguardian) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/BPMN2DataServiceImpl.java:163]: it is not the same thing.

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/ProcessAssetDesc.java [hunk 5]
```diff
@@ -39,6 +41,7 @@
 
     private String id;
     private String name;
+    private Map<String, Set<String>> tagsByVariable;
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/ProcessAssetDesc.java:44]: added

**Comment by afalhambra** [comment]
@elguardian seems there are some related failures, it also fails locally, though you need to run the whole jbpm-test-coverage module to get the error.

**Pull request merged**